### PR TITLE
CB-17871 Add Jackson compatibility checker for Consumption Service

### PR DESCRIPTION
**Pull request by bergerdenes**

### CODE REVIEW

File: cloud-consumption/src/main/java/com/sequenceiq/consumption/flow/consumption/storage/event/StorageConsumptionCollectionEvent.java [hunk 1]
```diff
@@ -11,7 +14,13 @@ public StorageConsumptionCollectionEvent(String selector, Long resourceId, Strin
         super(selector, resourceId, resourceCrn);
```
Comment (nadamdb) [inline cloud-consumption/src/main/java/com/sequenceiq/consumption/flow/consumption/storage/event/StorageConsumptionCollectionEvent.java:14]: shouldn't the same be added to this constructor as well?

File: cloud-consumption/src/main/java/com/sequenceiq/consumption/flow/consumption/storage/event/SendStorageConsumptionEvent.java [hunk 2]
```diff
@@ -14,8 +17,14 @@ public SendStorageConsumptionEvent(String selector, Long resourceId, String reso
         this.storageConsumptionResult = storageConsumptionResult;
```
Comment (nadamdb) [inline cloud-consumption/src/main/java/com/sequenceiq/consumption/flow/consumption/storage/event/SendStorageConsumptionEvent.java:17]: shouldn't the same be added to this constructor as well?

File: cloud-consumption/src/main/java/com/sequenceiq/consumption/flow/consumption/storage/event/StorageConsumptionCollectionHandlerEvent.java [hunk 3]
```diff
@@ -19,8 +22,15 @@ public StorageConsumptionCollectionHandlerEvent(String selector, Long resourceId
         this.storageConsumptionResult = storageConsumptionResult;
```
Comment (nadamdb) [inline cloud-consumption/src/main/java/com/sequenceiq/consumption/flow/consumption/storage/event/StorageConsumptionCollectionHandlerEvent.java:22]: shouldn't the same be added to this constructor as well?

File: cloud-consumption/src/main/java/com/sequenceiq/consumption/dto/StorageConsumptionResult.java [hunk 4]
```diff
@@ -1,10 +1,14 @@
...
 
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
 public class StorageConsumptionResult {
 
     private final double storageInBytes;
 
-    public StorageConsumptionResult(double storageInBytes) {
+    @JsonCreator
```
Comment (nadamdb) [inline cloud-consumption/src/main/java/com/sequenceiq/consumption/dto/StorageConsumptionResult.java:10]: just trying to understand the change, why is this added to this dto class and not to the other ones as well? (ConsumptionCreationDto, Credential)

File: cloud-consumption/src/main/java/com/sequenceiq/consumption/flow/consumption/storage/event/SendStorageConsumptionEvent.java (hunk 2 above)
Comment (SubhanHakverdiyev) [reply cloud-consumption/src/main/java/com/sequenceiq/consumption/flow/consumption/storage/event/SendStorageConsumptionEvent.java:17]: I think Jackson allow only one constructor to be creator that's why.

**Review by SubhanHakverdiyev — APPROVED** [review]

File: cloud-consumption/src/main/java/com/sequenceiq/consumption/dto/StorageConsumptionResult.java (hunk 4 above)
Comment (bergerdenes) [reply cloud-consumption/src/main/java/com/sequenceiq/consumption/dto/StorageConsumptionResult.java:10]: because it is referenced from a flow payload class (most probably one its fields is this type)

File: cloud-consumption/src/main/java/com/sequenceiq/consumption/flow/consumption/storage/event/SendStorageConsumptionEvent.java (hunk 2 above)
Comment (bergerdenes) [reply cloud-consumption/src/main/java/com/sequenceiq/consumption/flow/consumption/storage/event/SendStorageConsumptionEvent.java:17]: True, only one `@JsonCreator` is allowed
(ok, it can be hacked in a way, but let's just not consider that here :) )

File: cloud-consumption/src/main/java/com/sequenceiq/consumption/flow/consumption/storage/event/StorageConsumptionCollectionEvent.java (hunk 1 above)
Comment (bergerdenes) [reply cloud-consumption/src/main/java/com/sequenceiq/consumption/flow/consumption/storage/event/StorageConsumptionCollectionEvent.java:14]: nope

File: cloud-consumption/src/main/java/com/sequenceiq/consumption/flow/consumption/storage/event/StorageConsumptionCollectionHandlerEvent.java (hunk 3 above)
Comment (bergerdenes) [reply cloud-consumption/src/main/java/com/sequenceiq/consumption/flow/consumption/storage/event/StorageConsumptionCollectionHandlerEvent.java:22]: nope

**Review by nadamdb — APPROVED** [review]
Thanks for the clarifications, LGTM!
